File: class_ir.py
from span import SourceSpan


class ClassIR:
    def __init__(
        self,
        id: int,
        symbol_id: int,
        name: str,
        scope_id: int,       # parent scope
        body_scope_id: int,  # class-local scope
        bases: list,         # Base classes: Base, nn.Module, etc.
        decorators: list,
        span: SourceSpan,
    ):
        self.id = id
        self.symbol_id = symbol_id
        self.name = name
        self.scope_id = scope_id
        self.body_scope_id = body_scope_id
        self.bases = bases
        self.decorators = decorators
        self.span = span

File: rust/src/ir/stmt/class_ir.rs
use crate::ir::{expr_ir::ExprIR, span_ir::SourceSpan, stmt_ir::StmtIR};

#[derive(Debug, Clone)]
pub struct ClassIR {
    pub id: i64,
    pub symbol_id: i64,
    pub name: String,
    pub scope_id: i64,
    pub body_scope_id: i64,
    pub body: Vec<StmtIR>,
    pub bases: Vec<ExprIR>,
    pub decorators: Vec<ExprIR>,
    pub span: Option<SourceSpan>,
}
